bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/devices/ElroConnectsDeviceTemperatureSensor.java

```diff
@@ -0,0 +1,87 @@
+import org.openhab.binding.elroconnects.internal.handler.ElroConnectsBridgeHandler;
+import org.openhab.binding.elroconnects.internal.handler.ElroConnectsDeviceHandler;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ElroConnectsDeviceTemperatureSensor} is representing an Elro Connects temperature and humidity sensor
+ * device.
+ *
+ * @author Mark Herwege - Initial contribution
+ */
+@NonNullByDefault
+public class ElroConnectsDeviceTemperatureSensor extends ElroConnectsDevice {
+
+    private final Logger logger = LoggerFactory.getLogger(ElroConnectsDeviceTemperatureSensor.class);
+
+    public ElroConnectsDeviceTemperatureSensor(int deviceId, ElroConnectsBridgeHandler bridge) {
+        super(deviceId, bridge);
+    }
+
+    @Override
+    public void updateState() {
+        ElroConnectsDeviceHandler handler = getHandler();
+        if (handler == null) {
+            return;
+        }
+
+        ElroDeviceStatus elroStatus = ElroDeviceStatus.NORMAL;
+        int batteryLevel = 0;
+        int temperature = 0;
+        int humidity = 0;
+        String deviceStatus = this.deviceStatus;
+        if (deviceStatus.length() >= 8) {
+            batteryLevel = Integer.parseInt(deviceStatus.substring(2, 4), 16);
+            temperature = Byte.parseByte(deviceStatus.substring(4, 6), 16);
+            humidity = Integer.parseInt(deviceStatus.substring(6, 8));
+        } else {
+            elroStatus = ElroDeviceStatus.FAULT;
+            logger.debug("Could not decode device status: {}", deviceStatus);
+        }
+
+        switch (elroStatus) {
+            case FAULT:
+                handler.updateState(BATTERY_LEVEL, UnDefType.UNDEF);
+                handler.updateState(LOW_BATTERY, UnDefType.UNDEF);
+                handler.updateState(TEMPERATURE, UnDefType.UNDEF);
+                handler.updateState(HUMIDITY, UnDefType.UNDEF);
+                handler.updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, "Device " + deviceId + " has a fault");
+                break;
+            default:
+                handler.updateState(BATTERY_LEVEL, new DecimalType(batteryLevel));
+                handler.updateState(LOW_BATTERY, (batteryLevel < 15) ? OnOffType.ON : OnOffType.OFF);
+                handler.updateState(TEMPERATURE, new QuantityType<>(temperature, CELSIUS));
+                handler.updateState(HUMIDITY, new QuantityType<>(humidity, Units.PERCENT));
+                handler.updateStatus(ThingStatus.ONLINE);
+        }
+
+        logger.debug("Device ID {}, battery level {}, temperature {}, humidity {}", deviceId, batteryLevel, temperature,
```

Comment:
This is mostly a double up and should be considered to be removed to improve performance, a check is done to see what level the logger is set to. Any channels that get updated will have this already logged to the event.log